### ISSUE TITLE
Integration with Topola Genealogy Viewer

### ISSUE THREAD

**Issue by Wdavery** [gramps-project/Gramps.js#347]
In desktop Gramps, the [Interactive Family Tree ](https://gramps-project.org/wiki/index.php/Addon:Interactive_Family_Tree) addon ([source](https://github.com/gramps-project/addons-source/tree/master/Topola)) adds an integration with the interactive [Topola Genealogy Viewer](https://github.com/PeWu/topola-viewer).

Including it as an option within the Family Tree page would be valuable, as it provides a great way to view large and complex trees. 
I imagine it would be relatively straightforward to add since the integration with Gramps already exists.

**Comment by DavidMStraub** [comment]
Indeed! I think @PeWu and I were in contact a long time ago - I think this was still the predecessor project of Gramps Web - but it was never implemented. @PeWu would you be willing to support development of such an integration? Contributing would be awesome, but even if you don't have time for it, just being available for asking stupid questions would be helpful. 😄 

In general, our architecture is that we have custom elements, so we would have to wrap the Topola code into a custom element. Using typescript is fine because we are building with rollup.

The main open question for me would be in which format to fetch data. The API can create Gedcom or Gramps XML files, but it seems like an unnecessary complication to generate that in Python and parse it in JS again. The existing charts directly consume the Gramps objects in JSON form.

**Comment by PeWu** [comment]
Hi, can you give me a pointer where to look for the "Family Tree page" you are mentioning? I can definitely help with the integration.

The current way Topola Viewer is integrated with Gramps is that Gramps generates a GEDCOM file that is then parsed in Topola Viewer. This was the easiest way to integrate. Using an API that loads only needed data would be faster but requires some implementation on the Topola side.

My suggestion would be to start with integrating first using Gedcom just to see the result as quickly as possible. Then, we can consider using a more granular API to fetch only needed data.

**Comment by DavidMStraub** [comment]
Sounds good! The component holding the views for the graphs is here:
https://github.com/gramps-project/Gramps.js/blob/main/src/views/GrampsjsViewTree.js

Basically, adding a new view means adding a tab and a new `renderX` method that renders a new component (for isntance `<grampsjs-view-topola>`).

**Comment by DavidMStraub** [comment]
@geostag actually, Topola is quite similar to option 2 in #313 - I don't know if there can be some synergy. In principle, I wouldn't mind if we have several graphs with similar scope, it gives users more choice which is good...

**Comment by emyoulation** [comment]
Could a starting point just be installing the Topola Viewer as a parallel installation with GrampsWeb? GitHub says it uses TypeScript instead of Javascript. If Gramps Web was certain that it exists with every installation, does that simplify things?

https://github.com/PeWu/topola-viewer

**Comment by PeWu** [comment]
I created a first version of the Topola Viewer integration in PR #414.

Please take a look and tell me what you think.

**Comment by DavidMStraub** [comment]
If anyone wants to tackle this:

#414 was closed because it used iframes. Topola is an NPM package now, so it could be properly integrated, see https://github.com/gramps-project/gramps-web/pull/414#issuecomment-4213737352.
